### PR TITLE
fix frozen bots

### PR DESCRIPTION
**Pull request by ZappoMan**
after much debugging - I found the root cause of frozen bots... there was a subtle bug in our recording playback that would sometimes cause the frame timer to be set for an infinite time in the future.

Test plan:
- verify recording playback works without freezing. (note: freezing was random, but Anshuman has some tips on how to repro this)

### CODE REVIEW

File: libraries/recording/src/recording/Deck.cpp [hunk 1]
```diff
@@ -166,6 +166,12 @@ void Deck::processFrames() {
     if (!overLimit) {
         auto nextFrameTime = nextClip->positionFrameTime();
         nextInterval = (int)Frame::frameTimeToMilliseconds(nextFrameTime - _position);
+        if (nextInterval < 0) {
+            qCWarning(recordingLog) << " Unexected nextInterval < 0 nextFrameTime:" << nextFrameTime
```
Comment (thoys) [inline libraries/recording/src/recording/Deck.cpp]: Unexected -> Unexpected

Comment (thoys) [inline libraries/recording/src/recording/Deck.cpp]: Also an unexpected space on the left of the first word there

Comment (ZappoMan) [inline libraries/recording/src/recording/Deck.cpp]: damn it
:)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 086941356a842676ddb0ad05037df5f6f7edf6b5 available at: https://gist.github.com/251a332e98ce7c55fd645158dc2c8f28<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10321-0869413.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10321-0869413.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bace47af7c8e7ace57ba2bd1e56c32bdcdfa0303 available at: https://gist.github.com/251a332e98ce7c55fd645158dc2c8f28<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10321-bace47a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10321-bace47a.dmg

**Comment by anshuman64** [comment]
QA Pass
Tested in my Sandbox on my home PC. Had 30 bots running for 30 minutes and all were performing very well. Incredible! Merging this PR.
However, other problems with Record are surfacing that are not related to this PR. For example, recordings did not start playing after try to make on after installing the PR. Had to restart interface and make a 4th recording--and only then the other 3 retroactively played. Will create FogBugz tickets for these types of problems so we can try to repro them and tackle them individually.

**Pull request merged**